E2e testing sprint 4

EyalAzimov opened the pull request.

EyalAzimov committed:
[#15][End-to-end testing]
- created cypress test file

EyalAzimov committed:
[#15][End-to-end testing]
- added test to confirm event for beginning of spring break
- added test to confirm event for deadline to withdrawal from winter semester

EyalAzimov committed:
[#15][End-to-end testing]
- deleted 6 seconds of wait() in e2e tests

kikkiii21 commented on cypress/e2e/29_courseStudyHourEstimator.cy.js:
```diff
@@ -7,20 +7,17 @@ describe('Test Course Study Hour Estimator [SP-29]', function() {
     it('Check Study Hours', function(){
         cy.get('[data-test="navbar"]').click();
         cy.get('[data-test="Progress Report"]').click();
         //added data-test in CommonResources.js, ProfressReportHome.js
```
please remove commented code

EyalAzimov committed:
[#15][End-to-end testing]
- deleted commented code

EyalAzimov committed:
Merge branch 'main' of https://github.com/based-kwl/SemesterPlanner-client into e2e_testing_sprint_4

EyalAzimov committed:
[#15][Refactor]
- update with main

EyalAzimov committed:
[#15][Refactor]
- update with main

mnashat committed:
[#162][Testing]
- fixed failing tests; tests should pass more consistently/be less flaky (hopefully)

EyalAzimov committed:
Merge branch 'main' of https://github.com/based-kwl/SemesterPlanner-client into e2e_testing_sprint_4

 Conflicts:
	package-lock.json

EyalAzimov committed:
Merge branch 'main' of https://github.com/based-kwl/SemesterPlanner-client into e2e_testing_sprint_4

EyalAzimov committed:
[#15][Bug]
- modified aspect ratio during test because element was not clickable

EyalAzimov committed:
[#15][Bug]
- modified aspect ratio during test because element was not clickable
